ldrshared/validation/ldrschemas/jsonschemawrapper.py

```diff
@@ -0,0 +1,27 @@
+from jsonschema import validate, ValidationError
+
+from typing import Optional
+
+
+def confirm_indexer_trigger_message(message_dict: dict) -> Optional[ValidationError]:
+    """
+    Validates dict against required schema of the message used to trigger the
+    linked-data-repository-indexer service
+    """
+
+    # Note: At a time of writing the expected behaviour is the message
+    # will point to an input file (a csvw) on an attached volume, hence
+    # "input_data_path", change as needed as our thinking develops.
+    
+    # IMPORTANT: if you do change this, make sure to flush the queue,
+    # changing what is and is not a valid message while there are messages
+    # in flight is probably to be avoided.
```

Comment:
I do wonder if we should change the topic name when we alter the schema. You could imagine having a topic per version of schema. We should probably support two versions of the schema at any one point in time since we can't update everything at exactly the same point in time; i.e. subscribe to two topics - one for the latest version and one for the last version so we can upgrade without throwing messages away.

Reply:
This one sent me down a rabbit hole.

So.... I've added a schema version attribute to the messages, which I _think_ makes sense, so we can _choose_ to impart awareness of schema version on a message when we push to the queue and we can _choose_ to check-for/use/valdiate-against-it when we pull.

I'm stressing _choose_ as I don't want to lock the client behaviour down unilaterally (in case thats a bad idea), though (this is the point) it should allow us to treat messages conditionally based on the schema version (so support legacy messages but from a singe queue).

I think it makes but its gnarly, thoughts welcomed.

Reply:
Another approach might be to use the `$schema` property to deliniate versions. In csvcubed we've got a schema located at `http://purl.org/csv-cubed/qube-config/v1.0` which is a soft-redirect to the underlying schema for that version.

Then benefit is that if the JSON message contains that key/value pair, it'll automatically get validation/intellisense is text editors *and* you can use the value of `$schema` to figure out the `version` of the schema its using - since different versions should have different distinct URIs.


```json
{
"$schema": "http://purl.org/csv-cubed/qube-config/v1.0",
"other": "properties..."
}
```